[toh-xinyi] iP

toh-xinyi opened the pull request:
### EeveeBot

>  “EEVEE!” – Eevee ([source](https://www.youtube.com/watch?v=LxV-kesf44c))

EeveeBot frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,
1. download it from [here](https://github.com/toh-xinyi/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

Features:

 

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

And it is **FREE**!

***

If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

toh-xinyi committed:
Complete Level-1

toh-xinyi committed:
Add ability to store text

toh-xinyi committed:
Ability to mark tasks as done or undone

toh-xinyi committed:
able to track todos, deadlines and events now

toh-xinyi committed:
deals with some incorrect inputs

toh-xinyi committed:
Replaced wildcard imports

toh-xinyi committed:
Save tasks to hard disk automatically

toh-xinyi committed:
Teach Duke to understand dates

toh-xinyi committed:
Merge branch 'branch-Level-7'

toh-xinyi committed:
Merge branch 'branch-Level-8'

germainelee02 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,103 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+// import java.util.Scanner;
```
Can remove the comment here if you don't intend to use this.

toh-xinyi replied:
Removed. Thank you!

toh-xinyi committed:
Refactor the code to extract out closely related code as classes

toh-xinyi committed:
Organize the classes into suitable duke package

toh-xinyi committed:
Merge remote-tracking branch 'origin/add-gradle-support'

toh-xinyi committed:
Add gradle

toh-xinyi committed:
Change method name from formattedDescription() to formatDescription()

toh-xinyi committed:
Add JUnit tests to test the behavior of the code

toh-xinyi committed:
Add JavaDoc comments to the majority of methods.

toh-xinyi committed:
Tweak the code to comply with a coding standard

SFCoding123 commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,36 @@
+        this.dueDate = LocalDate.parse(dueDate);
+    }
+
+//    protected static LocalDateTime convertToDateFormat(String date) {
```
It will be better to delete unwanted comment codes according to the coding standards. When needed there will be old version to reference from.

toh-xinyi replied:
Deleted. Thank you!

SFCoding123 commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,36 @@
+package duke;
```
Maybe can include package documentation also

toh-xinyi replied:
Okay! Thank you!

SFCoding123 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,159 @@
+
+    public static void main(String[] args){
+        new Duke("data/tasks.txt").run();
+//        try {
```
Maybe can remove this chunk of comments as well following the coding standards

toh-xinyi replied:
Removed. Thank you!

SFCoding123 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
maybe the comment style can be improved.

toh-xinyi replied:
I have updated with a JavaDoc comment. Thank you for the feedback :)

SFCoding123 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,41 @@
+package duke;
+
+public class Ui {
```
I like how you organise the print system together to allows for easier modification.

toh-xinyi replied:
Thank you!

jiasheng59 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,159 @@
+                    break;
+                case "mark":
+                    int indexOfTaskToMarkDone = Integer.parseInt(command.split(" ")[1]) - 1;
+                    Task markedTask = tasks.markTaskAsDone(indexOfTaskToMarkDone);
+                    ui.showMarkingTaskDone(markedTask);
+                    break;
+                case "unmark":
+                    int indexOfTaskToMarkUndone = Integer.parseInt(command.split(" ")[1]) - 1;
+                    Task unmarkedTask = tasks.markTaskAsUndone(indexOfTaskToMarkUndone);
+                    ui.showMarkingTaskUndone(unmarkedTask);
+                    break;
+                case "todo":
+                    Task newTodo = Parser.makeTodoFromCommand(command);
+                    tasks.addTask(newTodo);
+                    ui.showAddingNewTask(newTodo, tasks);
+                    break;
+                case "deadline":
+                    Task newDeadline = Parser.makeDeadlineFromCommand(command);
+                    tasks.addTask(newDeadline);
+                    ui.showAddingNewTask(newDeadline, tasks);
+                    break;
+                case "event":
+                    Task newEvent = Parser.makeEventFromCommand(command);
+                    tasks.addTask(newEvent);
+                    ui.showAddingNewTask(newEvent, tasks);
+                    break;
+                case "delete":
+                    int indexOfTaskToDelete = Integer.parseInt(command.split(" ")[1]);
+                    Task taskToDelete = tasks.deleteTask(indexOfTaskToDelete);
+                    ui.showDeletingTask(taskToDelete, tasks);
+                    break;
+                default:
+                    throw new DukeException("Command not recognised.");
+                }
+                if (typeOfCommand.equals("bye")) {
+                    break;
+                }
+                storage.updateTaskList(tasks);
+                ui.askForNextCommand();
+            }
```
Would it be more suitable if this part is moved to Parser class?

toh-xinyi replied:
I will take your comment into consideration. Thank you!

jiasheng59 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,49 @@
+        }
+    }
+
+    public static Task makeTodoFromCommand(String line) {
+        String taskName = line.substring(5);
+        return new ToDos(taskName);
+    }
+
+    public static Task makeDeadlineFromCommand(String line) {
+        String taskInfo = line.substring(9);
+        String taskName = taskInfo.split(" /by")[0];
+        String taskDeadline = taskInfo.split("/by ")[1];
+        return new Deadlines(taskName, taskDeadline);
+    }
+
+    public static Task makeEventFromCommand(String line) {
+        String taskInfo = line.substring(6);
+        String taskName = taskInfo.split(" /")[0];
+        String taskStart = taskInfo.split(" /")[1].substring(5);
+        String taskEnd = taskInfo.split(" /")[2].substring(3);
+        return new Events(taskName, taskStart, taskEnd);
+    }
```
I like how these three methods are nicely abstracted out

toh-xinyi replied:
Thank you!

jiasheng59 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,49 @@
+        boolean isDone = taskInfo[1].trim().equals("1");
+        String taskDescription = taskInfo[2].trim();
+        switch (taskType) {
+        case "T":
+            return new ToDos(taskDescription, isDone);
+        case "D":
+            String taskTime = taskInfo[3].trim();
+            return new Deadlines(taskDescription, taskTime, isDone);
+        case "E":
+            String startTime = taskInfo[3].trim();
+            String endTime = taskInfo[4].trim();
+            return new Events(taskDescription, startTime, endTime, isDone);
```
Feel like this part can be improved through polymorphism

toh-xinyi replied:
Hmm, could you elaborate?

jiasheng59 commented on src/main/java/duke/Deadlines.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    @Override
+    public String formatDescription() {
```
I feel like this is a good name for this method

toh-xinyi replied:
Thank you!

jiasheng59 reviewed (commented):
LGTM, the overall code structure is neat and well-organized :smile:

toh-xinyi committed:
Allow users to find tasks by keyword

toh-xinyi committed:
Merge branch 'branch-Level-9'

toh-xinyi committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/TaskList.java

toh-xinyi committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java

toh-xinyi committed:
Add a GUI to Duke

toh-xinyi committed:
Remove redundant method

toh-xinyi committed:
Remove magic numbers

toh-xinyi committed:
Move bulk of handling input from Duke to Parser class

toh-xinyi committed:
Swap Duke and User pictures

toh-xinyi committed:
Add Duke and User pictures

toh-xinyi committed:
Make constants static final

toh-xinyi committed:
Enhance dates conversion and storage

toh-xinyi committed:
Add an assertion in TaskList class

An assertion added in the addTask method allows for check of possible
bugs in code when adding a new task to an empty task list.

Let's check if the task list is empty when adding a new task to the task
list

Using assertions can detect potential bugs in code when an assertion
failure is detected

toh-xinyi committed:
Add an assertion in Parser class

An assertion added in the handleInput method ensures the task is marked
correctly when given a command to mark or unmark a particular task as
done.

Let's insert an assertion to check if the task given is marked or
unmarked as told to do.

Using assertions can detect potential bugs in code when an assertion
failure is detected

toh-xinyi committed:
Merge pull request #1 from toh-xinyi/branch-A-Assertions

Add an assertion in Parser class

toh-xinyi committed:
Improve code quality

Code needs to be of high quality to enhance readability. Code is
simplified so that it can be understood easily.

Let's
* Shorten long methods in Parser class
* Add methods to follow SLAP principle

Code quality is enhanced. :)

toh-xinyi committed:
Merge pull request #2 from toh-xinyi/branch-A-CodeQuality

Improve code quality

toh-xinyi committed:
Debug to ensure functionality in find command

toh-xinyi committed:
Prevent addition of new event if the event clashes with an event
currently in the task list.

toh-xinyi committed:
Merge branch 'branch-B-DetectAnomalies'

toh-xinyi committed:
Fix problem of long text in dialog box cutting off

toh-xinyi committed:
Add a slight delay to exit the GUI to allow user to read exit message

toh-xinyi committed:
Change name of bot from Duke to PokeBot

toh-xinyi committed:
Make size of display pictures smaller and font smaller

toh-xinyi committed:
Change PokeBot to EeveeBot and add a little personality to Eeveebot

toh-xinyi committed:
Add representative screenshot

toh-xinyi committed:
Fix problem with saving tasks in filepath and add background to GUI

toh-xinyi committed:
Reupload UI image

toh-xinyi committed:
Change font and tweak dialogbox of UI

toh-xinyi committed:
Edit user guide

toh-xinyi committed:
Make small changes to User Guide

toh-xinyi committed:
Fix problem with file saving

toh-xinyi committed:
Repackage Task classes

toh-xinyi committed:
Add new exception to handle task with no content

toh-xinyi committed:
Change class names:
Deadlines -> Deadline
Events -> Event
ToDos -> ToDo

toh-xinyi committed:
Test using the I/O redirection technique

toh-xinyi committed:
Redo Automated Text UI Testing

toh-xinyi committed:
Add new exception to handle when index of task to delete is out of
bounds

toh-xinyi committed:
Edit hello message

toh-xinyi committed:
Tweak the look of GUI

toh-xinyi committed:
Fix bug related to adding deadline with no name

toh-xinyi committed:
Add help command to access list of commands and slight change to
width of MainWindow

toh-xinyi committed:
Add new Command classes and package them in command package

toh-xinyi committed:
Change hello statement

toh-xinyi committed:
Tweak to follow code quality standard

toh-xinyi committed:
Change title name from duke to eeveebot
